src/classes/ShareAccountController.cls

```diff
@@ -0,0 +1,19 @@
+public with sharing class ShareAccountController {
+    @AuraEnabled
+    public static void createAccountShare(Id recordId, Id userOrGroupId) {
+        List<AccountShare> checkForAvailability = [
+                SELECT Id, AccountId,UserOrGroupId
+                FROM AccountShare
+                WHERE AccountId = :recordId
+                AND UserOrGroupId = :userOrGroupId
+        ];
+        if (checkForAvailability.isEmpty()) {
```

Comment:
if 'checkForAvailability' not empty, can be checked AccessLevel to objects and updated if record has less access then you provide.